imagepipeline/src/main/java/com/facebook/imagepipeline/core/ProducerSequenceFactory.java

```diff
@@ -192,8 +192,12 @@ private static void validateEncodedImageRequest(ImageRequest imageRequest) {
     } else if (UriUtil.isLocalResourceUri(uri)) {
       return getLocalResourceFetchSequence();
     } else {
+      String uriString = uri.toString();
+      if (!TextUtils.isEmpty(uriString)) {
+        uriString.substring(0, Math.min(30, uriString.length));
```

Comment:
Hmm, I wonder how did we end up with 30 in the first place... Uri's are usually long, and the extension is usually at the end of them.